xbmc/Util.cpp

```diff
@@ -172,16 +172,18 @@ std::string GetHomePath(const std::string& strTarget, std::string strPath)
   auto bufSize = GetCurrentDirectoryW(0, nullptr);
   if (bufSize > 0)
   {
-    auto buf = new wchar_t[bufSize];
-    if (0 != GetCurrentDirectoryW(bufSize, buf))
+    std::unique_ptr<wchar_t> buf(new wchar_t[bufSize]);
```

Comment:
Shouldn't this be `std::unique_ptr<wchar_t[]>` for arrays? (It doesn't make difference for PODs I believe, but it's nicer)